lib/mayaHydra/hydraExtensions/sceneIndex/registration.cpp

```diff
@@ -417,10 +417,30 @@ MayaHydraSceneIndexRegistry::~MayaHydraSceneIndexRegistry()
         MSceneMessage::removeCallback(_AfterOpenCBId);
     }
     _AfterOpenCBId = 0;
+    _RemoveAllSceneIndexNodes();
     _registrationsByObjectHandle.clear();
     _registrations.clear();
 }
 
+void MayaHydraSceneIndexRegistry::_RemoveAllSceneIndexNodes()
```

Comment:
I fixed that function which had an issue when iterating on an array while removing its elements.

Reply:
To reduce code duplication, could we still call `_RemoveSceneIndexForNode`, just keeping the new way of always removing the first element? Something like :

```
while (_registrationsByObjectHandle.begin() != _registrationsByObjectHandle.end()) {
    _RemoveSceneIndexForNode(_registrationsByObjectHandle.begin()->first);
}
```

Reply:
Ha I see. But I did it like that on purpose to avoid calling _RemoveSceneIndexForNode from a MObjectHandle as from that we try to find it in the array then we take the iterator and we already have all of this in _RemoveAllSceneIndexNodes. So it would be a waste of time.

Reply:
Yeah that's why I think we should strive to re-use `_RemoveSceneIndexForNode` unless there is a functional limitation, having practically the same code in both makes me worried that in the future we might make a change in one but forget the other

Reply:
You're right in the sense that this code is not critical in term of speed, so let's do as your propose. Thanks.